errors.go

```diff
@@ -4,12 +4,17 @@ import "fmt"
 
 // Error is wrapper around error returned by Tarantool.
 type Error struct {
-	Code uint32
-	Msg  string
+	Code         uint32
+	Msg          string
+	ExtendedInfo *BoxError
```

Comment:
err.Code == err.ExtendedInfo.Code?
err.Msg == err.ExtendedInfo.Msg?

Does it make sense?

```Go
type Error struct {
    Code uint32
    Msg   string
}

type BoxError struct {
    Error
    // other fields
}
```

?

Reply:
> err.Code == err.ExtendedInfo.Code?
> err.Msg == err.ExtendedInfo.Msg?

It is expected to (but documentation doesn't say anything).

> Does it make sense?
I think so. Should we start to return BoxError instead of Error since `BoxError` implements Error interface? I think type casts may start to fail.

Reply:
Oh, we can not do it anyway (field and method with the same name Error):

https://go.dev/play/p/-xbX5TYbgX5

So don't worry about it.